src/proxy.js

```diff
@@ -0,0 +1,20 @@
+export default function getProxy (client) {
+  function getMethodProxy (service) {
+    const methodProxy = new Proxy({}, {
+      get: function (target, name) {
+        return function (data, callback) {
+          return client.call(service, name, data, callback)
+        }
+      }
+    })
+    return methodProxy
+  }
+
+  const serviceProxy = new Proxy({}, {
+    get: function (target, name) {
```

Comment:
only question here is, does this work with try/catch and our previous double callback issues etc? I'm not familiar with `Proxy`